### ISSUE TITLE
v15.0.1 upgrading from v14 'DeepLinkDelegate' is ambiguous for type lookup in this context

### ISSUE THREAD

**Issue by krisanthony** [urbanairship/ios-library#294]
# Preliminary Info

### What Airship dependencies are you using?
Airship 15.0.1

### What are the versions of any relevant development tools you are using?
Xcode 13.0

# Report

### What unexpected behavior are you seeing?
Receiving an error message when attempting to conform UIApplicationDelegate to `DeepLinkDelegate` - "'DeepLinkDelegate' is ambiguous for type lookup in this context". I cannot use `Airship.DeepLinkDelegate` either as Airship is a concrete class, so then receive the message "`DeepLinkDelegate` is not a member type of class 'Airship.Airship'". There are also no other classes etc. that conflict with this name in my project. 

### What is the expected behavior?
No error, able to conform to the protocol.

### What are the steps to reproduce the unexpected behavior?
Attempt to conform UIApplicationDelegate to `DeepLinkDelegate`.

### Do you have logging for the issue?
No.

**Comment by rlepinski** [comment]
How are you installing the framework? Pods or SPM?

**Comment by rlepinski** [comment]
I am not sure why its ambiguous for you, but there is an underlying issue with cocopoads that we introduced where the name Airship class conflicts with the framework... not sure the best way to resolve this without releasing another major SDK release. A quick workaround (hopefully) would be to create a new file with the following:
```
import Airship

public typealias AirshipDeepLinkDelegate = DeepLinkDelegate
```

Then use `AirshipDeepLinkDelegate` instead of `DeepLinkDelegate`. I will discuss with the team the best path forward for us.  It will most likely be either to rename the module for cocopoads to something like `AirshipFramework` or to revert the name of `Airship` back to `UAirship`. Sorry for the troubles.

**Comment by krisanthony** [comment]
Good idea there - thanks - now dealing with another issue which is the app crashes on launching. With `Thread 1: EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)`, digging into this but all I see in the stack trace is its due to accessing Airship.push - I am guessing it is not ready to be accessed yet - however we did not have an issue prior to this update - anything changed to be aware of that could be causing this?

**Comment by rlepinski** [comment]
When we moved from obj-c to swift we can no longer expose anything off Airship as null unspecified. We had to either make everything nullable or nonnull and assert if takeOff has not been called. We decided to go the nonnull path since we sometimes see customers calling things Airship before takeOff which led to undetected implementation issues.

Make sure to call takeOff before you access any of the singletons. If you have code that might not be ready or not you can check `isFlying` before access Airship:

```
guard Airship.isFlying else { return }
// or
if Airship.isFlying {

}
```

**Comment by rlepinski** [comment]
We are going to release SDK 16 hopefully this week where we change the module import for Cocoapods users to `AirshipKit` instead of `Airship`. Everything else will be the same, but it will require some find/replace in a project.

**Comment by krisanthony** [comment]
Yeah that is the funny thing, we are calling `takeOff` before, I will have to do some more investigation when time permits, but even breakpoints were not hitting, which was odd.

**Comment by rlepinski** [comment]
The only thing I can think of is maybe you are calling Airship before applicationDidFinishLaunching or maybe your config is invalid? https://github.com/urbanairship/ios-library/blob/main/Airship/AirshipCore/Source/Airship.swift#L192

**Comment by krisanthony** [comment]
We are calling in willFinish, I moved it to didFinish but did not seem to have an effect - still crashed on launch with that same error. We are using the default config. I'll be digging more into the issue this the week, appreciate your help so far.

**Comment by rlepinski** [comment]
Maybe another framework is trying to access it on a class loader? Do you use mparticle, segment, radar, or apptimize? There might be others that call through to our SDK.

**Comment by krisanthony** [comment]
Oh yeah we do use apptimize

**Comment by rlepinski** [comment]
I was able to reproduce it. Weird stack trace since it doesnt show apptimize at all. Ill escalate this and try to get a fix out ASAP.

**Comment by krisanthony** [comment]
💯 that's exactly what I was seeing - thanks for jumping on this so quickly

**Comment by rlepinski** [comment]
SDK 16 is out with the Airship -> AirshipKit change. Working with the Apptimize team to get a patch out for the Apptimize SDK

**Comment by rlepinski** [comment]
Apptimize issue should be fixed in 3.5.3

**Comment by krisanthony** [comment]
Great, thank you - seeing another issue with Apptimize integration. Though it is not related to iOS15 - would you prefer I open a new issue?
